lib/ui/painting/multi_frame_codec.cc

```diff
@@ -107,24 +107,30 @@ sk_sp<DlImage> MultiFrameCodec::State::GetNextFrameImage(
   std::optional<unsigned int> prior_frame_index = std::nullopt;
 
   if (requiredFrameIndex != SkCodec::kNoFrame) {
+    // We currently assume that frames can only ever depend on the immediately
+    // previous frame, if any. This means that
+    // `DisposalMethod::kRestorePrevious` is not supported.
     if (lastRequiredFrame_ == nullptr) {
-      FML_LOG(ERROR) << "Frame " << nextFrameIndex_ << " depends on frame "
-                     << requiredFrameIndex
-                     << " and no required frames are cached.";
-      return nullptr;
+      FML_DLOG(INFO)
+          << "Frame " << nextFrameIndex_ << " depends on frame "
+          << requiredFrameIndex
+          << " and no required frames are cached. Using blank slate instead.";
     } else if (lastRequiredFrameIndex_ != requiredFrameIndex) {
       FML_DLOG(INFO) << "Required frame " << requiredFrameIndex
-                     << " is not cached. Using " << lastRequiredFrameIndex_
-                     << " instead";
-    }
-
-    if (lastRequiredFrame_->getPixels() &&
-        CopyToBitmap(&bitmap, lastRequiredFrame_->colorType(),
-                     *lastRequiredFrame_)) {
-      prior_frame_index = requiredFrameIndex;
+                     << " is not cached. Using blank slate instead.";
+    } else {
```

Comment:
I think this is a bug. Will follow up with some more details shortly.

Reply:
For GIFs at least, this isn't safe. I vaguely remember discussing this before a while back. Something like the current frame can end up depending on the last frame and the framework doesn't draw both the last and current frame so it needs a complete frame.

Reply:
(IOW, the GIF can require a last frame that was several frames ago but still needs to be painted)

Reply:
Is the SkBitmap allocation not getting initialized with zeroes? This logic change should make the copying the action of copying the last frame more restrictive, not less. We were copying the previous frame's contents in cases where we're supposed to composite atop a blank frame, which was visible in this gif as well: https://github.com/flutter/flutter/issues/61150#issuecomment-679055858